src/mongocxx/test/gridfs/bucket.cpp

```diff
@@ -857,18 +857,105 @@ TEST_CASE("gridfs::bucket::download_to_stream works", "[gridfs::bucket]") {
     db["fs.files"].drop();
     db["fs.chunks"].drop();
 
-    std::int64_t length = 10;
+    std::int64_t length = 19;
     std::int32_t chunk_size = 4;
     bsoncxx::types::bson_value::view id{bsoncxx::types::b_oid{bsoncxx::oid{}}};
-    auto expected_bytes = manual_gridfs_initialize(db, length, chunk_size, id);
-
+    auto file_bytes = manual_gridfs_initialize(db, length, chunk_size, id);
     std::ostringstream os;
-    bucket.download_to_stream(id, &os);
 
-    auto str = os.str();
-    std::vector<std::uint8_t> actual_bytes{str.begin(), str.end()};
+    SECTION("download complete file") {
+        bucket.download_to_stream(id, &os);
+
+        auto str = os.str();
+        std::vector<std::uint8_t> actual_bytes{str.begin(), str.end()};
+
+        REQUIRE(file_bytes == actual_bytes);
+    }
+
+    SECTION("download complete file with start/end offsets") {
+        bucket.download_to_stream(id, &os, 0, length);
+
+        auto str = os.str();
+        std::vector<std::uint8_t> actual_bytes{str.begin(), str.end()};
+
+        REQUIRE(file_bytes == actual_bytes);
+    }
+
+    SECTION("download partial file") {
+        auto check_downloaded_content = [&](std::size_t start, std::size_t end) {
+            bucket.download_to_stream(id, &os, start, end);
+
+            auto str = os.str();
+            std::vector<std::uint8_t> actual_bytes{str.begin(), str.end()};
+            REQUIRE(actual_bytes.size() == end - start);
+            const std::vector<std::uint8_t> expected_bytes{file_bytes.begin() + start,
+                                                           file_bytes.begin() + end};
+            REQUIRE(expected_bytes == actual_bytes);
+        };
+
+        SECTION("whole file") {
+            check_downloaded_content(0, length);
+        }
+
+        SECTION("no data") {
+            check_downloaded_content(0, 0);
+        }
+
+        SECTION("at file start") {
+            SECTION("partial chunk") {
+                check_downloaded_content(0, chunk_size - 1);
+            }
+
+            SECTION("complete chunk") {
+                check_downloaded_content(0, chunk_size);
+            }
+
+            SECTION("across chunks") {
+                check_downloaded_content(0, 2 * chunk_size);
+            }
+        }
+
+        SECTION("middle of the file") {
+            SECTION("partial chunk") {
+                const auto start = chunk_size+ chunk_size / 2;
```

Comment:
I'm getting some errors:
```
~/projects/mongo-cxx-driver$ python2 ./etc/clang_format.py format
Downloading clang-format 7.0.1 from https://s3.amazonaws.com/boxes.10gen.com/build/clang-format-7.0.1-rhel70.tar.gz, saving to /tmp/temp.tar.gz
/home/ogrech/projects/mongo-cxx-driver/build/clang-format-7.0.1: error while loading shared libraries: libtinfo.so.5: cannot open shared object file: No such file or directory
WARNING: clang-format found in path, but incorrect version found at /home/ogrech/projects/mongo-cxx-driver/build/clang-format-7.0.1 with version: clang-format call failed.
ERROR: exiting because of previous warning.
```